src/backends/perf_pt/test_helpers.c

```diff
@@ -0,0 +1,144 @@
+//
+// Subject to the condition set forth below, permission is hereby granted to any
+// person obtaining a copy of this software, associated documentation and/or
+// data (collectively the "Software"), free of charge and under any and all
+// copyright rights in the Software, and any and all patent rights owned or
+// freely licensable by each licensor hereunder covering either (i) the
+// unmodified Software as contributed to or provided by such licensor, or (ii)
+// the Larger Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#define _GNU_SOURCE
+
+#include <link.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <inttypes.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/auxv.h>
+#include <err.h>
+#include <elf.h>
+
+#include "perf_pt_private.h"
+
+// Arguments for perf_pt_append_self_ptxed_raw_args().
+struct append_args {
+    int vdso_fd;
+    char *vdso_filename;
+    void *rs_args_vec;
+};
+
+// Public prototypes.
+bool perf_pt_append_self_ptxed_raw_args(struct append_args *);
+
+// Private prototypes.
+static int append_self_ptxed_raw_args_cb(struct dl_phdr_info *, size_t, void *);
+
+// Rust functions.
+extern void push_ptxed_arg(void *, const char *);
+
+// More than enough space for a ptxed `--raw` argument parameter.
+#define MAX_RAW_ARG 128
+
+/*
+ * Walks the program headers pushing ptxed `--raw` arguments for the current
+ * process into the supplied (pointer to a) Rust `AppendSelfPtxedArgs`
+ * instance.
+ */
+bool
+perf_pt_append_self_ptxed_raw_args(struct append_args *rs_args)
+{
+    if (dl_iterate_phdr(append_self_ptxed_raw_args_cb, rs_args) == 1) {
```

Comment:
Can just be `return dl_iterate_phdr(...) != 1;`

Reply:
Yeah. Don't know why I made that so complicated. It probably had some other code around it at some point.